index.d.ts

```diff
@@ -0,0 +1,4 @@
+type Error = string;
+
+export function whois(domain: string): string;
+export function resolveMxRecords(domain: string): Array<string> | Error;
```

Comment:
Doesn't seem to be the right return type no ?

For me in case of error you'r returning an Object `error` and `context` property.